dali/pipeline/operators/reader/nvdecoder/nvdecoder.cc

```diff
@@ -142,42 +145,54 @@ int NvDecoder::handle_display(void* user_data,
   return decoder->handle_display_(disp_info);
 }
 
+// Prepare params and calls cuvidCreateDecoder
 int NvDecoder::handle_sequence_(CUVIDEOFORMAT* format) {
   frame_base_ = {static_cast<int>(format->frame_rate.denominator),
                   static_cast<int>(format->frame_rate.numerator)};
-
-  // Prepare params and calls cuvidCreateDecoder
-  return decoder_.initialize(format);
+  int ret = kNvcuvid_failure;
+  try {
+    ret = decoder_.initialize(format);
+  } catch (...) {
+    done_ = true;
+    captured_exception_ = std::current_exception();
+    // Main thread is waiting on frame_queue_
+    frame_queue_.shutdown();
+  }
+  return ret;
 }
 
 int NvDecoder::handle_decode_(CUVIDPICPARAMS* pic_params) {
   int total_wait = 0;
   constexpr auto sleep_period = 500;
   constexpr auto timeout_sec = 20;
   constexpr auto enable_timeout = false;
+
+  // If something went wrong during init we exit directly
+  if (done_) return kNvcuvid_failure;
```

Comment:
It is a bit surprising that done_ == true means failure.

Reply:
Hmm, this `kNvcuvid_failure` is telling this to the decoder "something went wrong is this callback, so stop further processing". We can rename `done_` to `stop_` to have something generic

Reply:
stop_ sounds better